metaflow/client/core.py

```diff
@@ -1717,6 +1747,83 @@ def end_task(self):
 
         return end_step.task
 
+    def add_tag(self, tag):
```

Comment:
This is the issue compared to the current version we have. This version makes more sense but may break a bunch of our flows. For consistency, we may be able to break our users but maybe it would be better to add an error message to indicate what changed. I am not a fan of this solution but I understand the desire to give the function a better name. And it was what I was going for in 547.